crates/katana/storage/provider/src/providers/fork/backend.rs

```diff
@@ -705,6 +837,261 @@ mod tests {
         assert_eq!(stats, 5, "Backend should have 5 ongoing requests.")
     }
 
+    #[test]
+    fn get_nonce_request_should_be_deduplicated() {
+        // start a mock remote network
+        start_tcp_server("127.0.0.1:8081".to_string());
+
+        let handle = create_forked_backend("http://127.0.0.1:8081", 1);
+
+        // check no pending requests
+        let stats = handle.stats().expect(ERROR_STATS);
+        assert_eq!(stats, 0, "Backend should not have any ongoing requests.");
+
+        // send requests to the backend
+        let h1 = handle.clone();
+        thread::spawn(move || {
+            h1.get_nonce(felt!("0x1").into()).expect(ERROR_SEND_REQUEST);
+        });
+        let h2 = handle.clone();
+        thread::spawn(move || {
+            h2.get_nonce(felt!("0x1").into()).expect(ERROR_SEND_REQUEST);
+        });
+        // Different request, should be counted
+        let h3 = handle.clone();
+        thread::spawn(move || {
+            h3.get_nonce(felt!("0x2").into()).expect(ERROR_SEND_REQUEST);
+        });
+
+        // wait for the requests to be handled
+        thread::sleep(Duration::from_secs(1));
+
+        // check request are handled
+        let stats = handle.stats().expect(ERROR_STATS);
+        assert_eq!(stats, 2, "Backend should only have 2 ongoing requests.")
+    }
```

Comment:
Let's do a check before sending the third request to make sure the requests actually get dedup'd. Having it only at the end can be a bit vague because it doesn't indicate whether the 2 ongoing ones are exactly what we're expecting. 

```suggestion
        // send requests to the backend
        let h1 = handle.clone();
        thread::spawn(move || {
            h1.get_nonce(felt!("0x1").into()).expect(ERROR_SEND_REQUEST);
        });
        let h2 = handle.clone();
        thread::spawn(move || {
            h2.get_nonce(felt!("0x1").into()).expect(ERROR_SEND_REQUEST);
        });
        
        // wait for the requests to be handled
        thread::sleep(Duration::from_secs(1));

        // check request are handled
        let stats = handle.stats().expect(ERROR_STATS);
        assert_eq!(stats, 1, "Backend should only have 1 ongoing requests.")
        
        // Different request, should be counted
        let h3 = handle.clone();
        thread::spawn(move || {
            h3.get_nonce(felt!("0x2").into()).expect(ERROR_SEND_REQUEST);
        });

        // wait for the requests to be handled
        thread::sleep(Duration::from_secs(1));

        // check request are handled
        let stats = handle.stats().expect(ERROR_STATS);
        assert_eq!(stats, 2, "Backend should only have 2 ongoing requests.")
    }
```